### DISCUSSION TITLE
How to define linear form when using Argyris element?

### DISCUSSION THREAD

**Discussion by JunmingDuan** [firedrakeproject/firedrake#2574] [Firedrake support] [answered]
Dear all,

I want to solve the Helmholtz equation  - \Delta u -k^2*u = f  using the Argyris element.
The linear form is \int f*v dx.
In the tutorials, I learned that f is interpolated to the finite element space, then just use inner(f, v) * dx to get the linear form.
But the interpolation is not supported by Argyris element.
So how to define the linear form in this case?
Thanks for any help!

**Comment by colinjcotter** [comment, marked as answer]
Hi Junming,
  In this variational problem, f can be from another finite element space, or it can be a UFL expression directly.

all the best
cjc

On 8 Oct 2022, at 14:26, Junming Duan ***@***.***> wrote:

﻿

Dear all,

I want to solve the Helmholtz equation - \Delta u -k^2u = f using the Argyris element.
The linear form is \int fv dx.
In the tutorials, I learned that f is interpolated to the finite element space, then just use inner(f, v) * dx to get the linear form.
But the interpolation is not supported by Argyris element.
So how to define the linear form in this case?
Thanks for any help!
